go-controller/pkg/clustermanager/secondary_network_cluster_controller.go

```diff
@@ -0,0 +1,50 @@
+package clustermanager
+
+import (
+	"context"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+// secondaryNetworkClusterController is the object for managing the secondary
+// layer3 networks for all the nodes. Listens to the node events.
+type secondaryNetworkClusterController struct {
```

Comment:
You mean when a secondary network is created,  the function NewNetworkController (of secondaryNetworkClusterManager) can return an instance of networkClusterController directly instead of this wrapper - secondaryNetworkClusterController  ?

If so,  we cannot do so since NewNetworkController returns an object of type - nad.NetworkController and some of the missing interface functions needs to be implemented in networkClusterController.

And hence this wrapper struct secondaryNetworkClusterController which also embeds util.NetInfo and util.NetConfInfo.